engineyard/ey-cookbooks-stable-v6
# [FB-863] Remove hardcoded apt package versions

Description of your patch
-------------
This PR removes hard-coded apt package versions from the cookbooks. This is necessary in order to allow for Ubuntu package updates to be applied and to avoid errors when a package would be downgraded.

Recommended Release Notes
-------------
* Remove hard-coded apt package version restrictions, therefore adhering to the Ubuntu package update policy.
* Fix a bug which disabled TLS v1.2 for nginx.
* Fix a bug in parsing the Redis version string when installing from source. 

Estimated risk
-------------
High. This is a significant change, affecting many different recipes. Therefore it needs to be tested carefully with different configurations.

Components involved
-------------
* redis
* nginx
* collectd
* ruby
* newrelic_infra
* memcached
* tinyproxy
* haproxy
* syslog-ng
* run-one
* packages
* openssl
* monit
* ey-lib
* ey-base
* unattended-upgrades
* passenger5
* ntp
* nodejs
* mysql
* ey-init
* env_vars

Description of testing done
-------------
1. Booted a custom environment with:
    * 1 app instance
    * 2 db instances (master + slave)
    * utility instances for: redis, memcached, tinyproxy
2. Activated unattended upgrades
3. Used passenger5 as app server
4. Used the redis, memcached, tinyproxy, and newrelic_infra custom chef recipes
5. Booted the environment and checked if apply succeeds
6. Deployed the todo application and verified if it works.

QA Instructions
-------------
* Test solo, staging, and production environments
* Test different app servers
* Test different databases
* Test an application with sidekiq
* Activate unattended upgrades, keep the env running for five days, run apply every day and verify that each one is successful
* Test with an App Load Balancer
* Test with a SSL certificate, terminated in nginx (verify that TLS v1.2 is accepted)
